lib/init/grass.py

```diff
@@ -2213,7 +2213,7 @@ def print_params(params):
             try:
                 revision = linerev.split(" ")[1]
                 sys.stdout.write("%s\n" % revision[1:])
-            except:
+            except (IndexError, TypeError, AttributeError):
```

Comment:
AttributeError - Generally, attribute error should not be caught unless there is a specific reason to do so (like dealing with different types of objects and reacting to the differences in their attributes).

TypeError - Where this would happen and is that something we want to hide or cause traceback?

Reply:
TypeError happens as following. 

TypeError: This exception could be raised by revision[1:] if revision is not a string or bytes-like object. The slicing operation is not supported by all data types, so if revision is not of a type that supports slicing, a TypeError would be raised.

Reply:
Under which conditions revision would not be a string? The revision value is a result of the line 2214 which splits a string using [str.split](https://docs.python.org/3.3/library/stdtypes.html#str.split) which returns _list of the words in the string_.